pymc_marketing/clv/models/basic.py

```diff
@@ -78,31 +137,151 @@ def _fit_MAP(self, **kwargs):
         map_strace.record(map_res)
         map_strace.close()
         trace = MultiTrace([map_strace])
-        return pm.to_inference_data(trace, model=model)
+        idata = pm.to_inference_data(trace, model=model)
+        self.set_idata_attrs(idata)
+        self.idata = idata
+        return self.idata
+
+    @classmethod
+    def load(cls, fname: str):
+        """
+        Creates a ModelBuilder instance from a file,
+        Loads inference data for the model.
+
+        Parameters
+        ----------
+        fname : string
+            This denotes the name with path from where idata should be loaded from.
+
+        Returns
+        -------
+        Returns an instance of ModelBuilder.
+
+        Raises
+        ------
+        ValueError
+            If the inference data that is loaded doesn't match with the model.
+        Examples
+        --------
+        >>> class MyModel(ModelBuilder):
+        >>>     ...
+        >>> name = './mymodel.nc'
+        >>> imported_model = MyModel.load(name)
+        """
+        filepath = Path(str(fname))
+        idata = az.from_netcdf(filepath)
+        dataset = idata.fit_data.to_dataframe()
+
+        model = cls(
+            dataset,
+            model_config=json.loads(idata.attrs["model_config"]),  # type: ignore
+            sampler_config=json.loads(idata.attrs["sampler_config"]),
+        )
+        model.idata = idata
+
+        model.build_model()
+        # All previously used data is in idata.
+
+        return model
+
+    @staticmethod
+    def _check_prior_ndim(prior, ndim=0):
+        if prior.ndim != ndim:
+            raise ValueError(
+                f"Prior variable {prior} must be have {ndim} ndims, but it has {prior.ndim} ndims."
+            )
+
+    def create_distribution_from_prior(self, name: str, **kwargs) -> TensorVariable:
```

Comment:
This name doesn't make much sense. Perhaps `create_distribution_from_dict`? Also I think it can be private with a leading underscore. Users are unlikely to care about this.

Reply:
Also should be a staticmethod AFAICT

Reply:
create_distribution_from_dict makes even less sense as we're not passing a dict to the function, maybe simply create_distribution? I'm not really seeing a reason why this should be static, it's not meant to be used outside of init, after the super() are initialised, so it will be available in child classes, and once again the basic shouldn't be initialised and used as a standalone class

Reply:
renamed, it's called now _create_distribution, a static method with the ndim param